File: packages/sdk-python/scripts/live_trace_smoke.py
from __future__ import annotations

import uuid

import rifft


def main() -> None:
    test_id = f"python-smoke-{uuid.uuid4().hex[:12]}"

    rifft.init(project_id="default", endpoint="http://localhost:4318")

    @rifft.trace(agent_id="researcher", framework="custom")
    def research(topic: str) -> dict[str, object]:
        with rifft.span("tool_call", agent_id="researcher", framework="custom") as span:
            span.set_attribute("trace.test_id", test_id)
            span.set_attribute("tool.name", "web_search")
            span.set_attribute("tool.input", {"query": topic})
            span.set_attribute("tool.output", {"hits": 3})
            span.set_attribute("tool.result_length", 3)
        return {"topic": topic, "status": "ok"}

    with rifft.span("python.smoke.run", agent_id="orchestrator", framework="custom") as span:
        span.set_attribute("trace.test_id", test_id)
        result = research("python live smoke")
        span.set_attribute("trace.result", result)

    with rifft.span("output.validate", agent_id="orchestrator", framework="custom") as span:
        span.set_attribute("validation.passed", result.get("status") == "ok")
        span.set_attribute("validation.output", result)

    provider = rifft.get_tracer_provider()
    shutdown = getattr(provider, "shutdown", None)
    if callable(shutdown):
        shutdown()

    print(test_id)


if __name__ == "__main__":
    main()

File: packages/sdk-js/scripts/live-trace-smoke.ts
import { init, trace, withSpan } from "../src/index.js";

const testId = `sdk-js-smoke-${Math.random().toString(16).slice(2, 10)}`;

init({
  project_id: "default",
  endpoint: "http://localhost:4318",
});

const researcher = trace({
  agent_id: "researcher",
  framework: "custom",
  span_name: "agent.research",
})(async function researcherAgent(query: string) {
  return withSpan("tool.call", { agent_id: "researcher", framework: "custom" }, async (span) => {
    span.setAttribute("tool.name", "web_search");
    span.setAttribute("tool.input", query);
    span.captureDecision({
      system_prompt: "Collect sources",
      conversation_history: [{ role: "user", content: query }],
      available_tools: ["web_search"],
      chosen_action: "web_search",
    });
    return ["source-a", "source-b"];
  });
});

const writer = trace({
  agent_id: "writer",
  framework: "custom",
  span_name: "agent.write",
})(async function writerAgent(sources: string[]) {
  return withSpan("agent.compose", { agent_id: "writer", framework: "custom" }, async (span) => {
    span.setAttribute("input", { sources });
    span.setAttribute("trace.test_id", testId);
    return `summary:${sources.length}`;
  });
});

const main = async () => {
  const result = await withSpan("sdk.js.smoke.root", { agent_id: "orchestrator", framework: "custom" }, async (span) => {
    span.setAttribute("trace.test_id", testId);
    const sources = await researcher(`Investigate ${testId}`);
    const summary = await writer(sources);
    await withSpan("output.validate", { agent_id: "orchestrator", framework: "custom" }, async (validationSpan) => {
      validationSpan.setAttribute("validation.passed", summary.startsWith("summary:"));
      validationSpan.setAttribute("validation.output", summary);
    });
    return summary;
  });

  console.log(testId);
  console.log(result);
};

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});
